packages/accounts/src/msca/base.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  BaseSmartContractAccount,
+  SignerSchema,
+  createBaseSmartAccountParamsSchema,
+  type BatchUserOperationCallData,
+  type SupportedTransports,
+} from "@alchemy/aa-core";
+import {
+  encodeFunctionData,
+  type Address,
+  type FallbackTransport,
+  type Hex,
+  type Transport,
+} from "viem";
+import { z } from "zod";
+import { IStandardExecutorAbi } from "./abis/IStandardExecutor.js";
+
+export const createModularSmartContractAccountSchema = <
+  TTransport extends SupportedTransports = Transport
+>() =>
+  createBaseSmartAccountParamsSchema<TTransport>().extend({
+    owner: SignerSchema,
+  });
+
+export type ModularSmartContractAccountParams = z.input<
+  ReturnType<typeof createModularSmartContractAccountSchema>
+>;
+
+export abstract class BaseModularSmartContractAccount<
```

Comment:
rohan's gonna kill me....

I think we should explore some kind of Builder Pattern for MSCA that lets you generate new classes by doing something like:

```
type MSCA = ISmartContractAccount & MSCASpecificMethods // like withPlugin?

export const createMultiOwnerMSCA = <R extends ISmartContractAccount>({owner}: MultiOwnerMSCAParams): R => { 
  return ModularSmartContractAccountBuilder()
    .withStandardExecutor()
    .withPlugin(MultiOwnerPlugin)
    .withSignMethods(account => ({
      signMessage: ...
      signTypedData: ...
      signUserOperationHash: ...
    }))
}
```

Reply:
Actually after talking to @adam-alchemy and @howydev, we will probably want something like the above. Right now, the standard executor function is only callable in context of the root owner plugin. So for session key plugins or alternative signers, they will have their own methods to call. So that means that `executeBatch` is not appropriate the way it is defined here.

We'll want to introduce some kind of MSCA class builder that returns an MSCA class that can then be further extended with decorators (from plugin gen)

Reply:
prototyped this some here: https://github.com/alchemyplatform/aa-sdk/pull/264